homeassistant/components/shelly/event.py

```diff
@@ -95,6 +103,21 @@ async def async_setup_entry(
                 async_remove_shelly_entity(hass, EVENT_DOMAIN, unique_id)
             else:
                 entities.append(ShellyRpcEvent(coordinator, key, RPC_EVENT))
+
+        script_instances = get_rpc_key_instances(
+            coordinator.device.status, SCRIPT_EVENT.key
+        )
+        for script in script_instances:
+            event_types = await get_rpc_script_event_types(
+                coordinator.device, int(script.split(":")[-1])
+            )
+            if not event_types:
+                continue
+
+            entities.append(
```

Comment:
I think we need to use `async_remove_orphaned_entities` function to remove orphaned entities (after the user removes the script or removes `emitEvent` from the script).